src/interactions/glue_fs.jl

```diff
@@ -0,0 +1,241 @@
+
+The Finnis-Sinclair interaction. This interaction expects units to be of 
+these https://lammps.sandia.gov/doc/units.html units (eV, Å, K, ps and so on).
+"""
+struct FinnisSinclair <: GlueInteraction
+    nl_only::Bool
+    element_pair_map::Dict
+    params::DataFrame
+end
+
+
+kb = 8.617333262145e-5 # eV/K
+
+"""
+    get_finnissinclair1984(nl_only)
+
+Finnis and Sinclair 1984 parameterization: https://doi.org/10.1080/01418618408244210
+"""
+function get_finnissinclair1984(nl_only::Bool)
+    
+    elements = ["V", "Nb", "Ta", "Cr", "Mo", "W", "Fe"]
+    element_pairings = [string(el, el) for el in elements]
+    element_pair_map = Dict(pair => i for (i,pair) in enumerate(element_pairings))
+    
+    df = DataFrame(
+        element_pair = element_pairings,
+        d = [3.692767, 3.915354, 4.076980, 3.915720, 
+            4.114825, 4.400224, 3.699579],
+        # (Å)
+        A = [2.010637, 3.013789, 2.591061, 1.453418, 
+            1.887117, 1.896373, 1.889846],
+        # (eV)
+        β = [0, 0, 0, 1.8, 0, 0, 1.8],
+        # (1)
+        c = [3.8, 4.2, 4.2, 2.9, 3.25, 3.25, 3.4],
+        # (Å)
+        c₀ = [-0.8816318, -1.5640104, 1.2157373, 29.1429813, 
+            43.4475218, 47.1346499, 1.2110601],
+        # (1)
+        c₁ = [1.4907756, 2.0055779, 0.0271471, -23.3975027, 
+            -31.9332978, -33.7665655, -0.7510840],
+        # (1)
+        c₂ = [-0.3976370, -0.4663764, -0.1217350, 4.7578297, 
+            6.0804249, 6.2541999, 0.1380773],
+        # (1)
+    )
+
+    masses = Dict("V" => 50.9415, "Nb" => 92.9064, "Ta" => 180.9479,
+                  "Cr" => 51.996, "Mo" => 95.94, "W" => 183.85,
+                  "Fe" => 55.847) # g/mole
+
+    # Å
+    bcc_lattice_constants = Dict(
+        "V" => 3.0399, "Nb" => 3.3008, "Ta" => 3.3058, 
+        "Cr" => 2.8845, "Mo" => 3.1472, "W" => 3.1652, 
+        "Fe" => 2.8665
+    )
+
+    
+    fs84 = FinnisSinclair(nl_only, element_pair_map, df)
+    
+    reference_energies = DataFrame(
+        element_pair = element_pairings,
+        u = [5.31, 7.57, 8.1, 4.1, 6.82, 8.9, 4.28],
+        u_vac = [1.92, 2.64, 3.13, 1.97, 2.58, 3.71, 1.77]
+    )
+    return fs84, elements, masses, bcc_lattice_constants, reference_energies 
+end
+
+"""
+    glue(r,β,d; f=10.)
+
+The core component of the Finnis-Sinclair type GlueInteraction. 
+Used to calculate contribution to the glue value of an atom, based on 
+its neighbouring atoms.
+f is a fudge factor to help translate a Å model to a nm model.
+"""
+function glue(r, β, d)
+    return r > d ? 0 : (r-d)^2 + β*(r-d)^3/d
+end
+
+"""
+    ∂glue_∂r(r, β, d; f=10.)
+
+Derivative of the glue density function.
+"""
+∂glue_∂r(r, β, d) = ForwardDiff.derivative(r -> glue(r,β,d), r)
+
+"""
+    Uglue(ρ, A)
+
+Energy based on the glue density .
+"""
+function Uglue(ρ, A)
+   return -A * √ρ 
+end
+
+
+"""
+    ∂Uglue_∂ρ(ρ, A)
+
+Energy derivative given glue density.
+"""
+∂Uglue_∂ρ(ρ,A) = ForwardDiff.derivative(ρ -> Uglue(ρ,A), ρ)
```

Comment:
This could be simply derived analytically

```suggestion
∂Uglue_∂ρ(ρ,A) = -A /(2 * √ρ)
```